sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -734,6 +734,24 @@ class SQLContext(@transient val sparkContext: SparkContext)
   def table(tableName: String): DataFrame =
     DataFrame(this, catalog.lookupRelation(Seq(tableName)))
 
+  /**
+   * Returns a [[DataFrame]] containing names of existing tables in the current database.
+   * The returned DataFrame has two columns, tableName and isTemporary (a column with BooleanType
+   * indicating if a table is a temporary one or not).
+   */
+  def tables(databaseName: String): DataFrame = {
```

Comment:
Really, `showTables` or `listTables` both sounds like a `Unit` function that prints to the screen (like the `show` function we already have on `DataFrame`).